local-cli/runAndroid/runAndroid.js

```diff
@@ -85,29 +85,47 @@ function buildAndRun(args) {
     ? 'gradlew.bat'
     : './gradlew';
 
-  const packageName = fs.readFileSync(
-      'app/src/main/AndroidManifest.xml',
-      'utf8'
-    ).match(/package="(.+?)"/)[1];
-
   const adbPath = getAdbPath();
   if (args.deviceId) {
     if (isString(args.deviceId)) {
-        runOnSpecificDevice(args, cmd, packageName, adbPath);
+        runOnSpecificDevice(args, cmd, adbPath);
     } else {
       console.log(chalk.red('Argument missing for parameter --deviceId'));
     }
   } else {
-    runOnAllDevices(args, cmd, packageName, adbPath);
+    runOnAllDevices(args, cmd, adbPath);
+  }
+}
+
+function fullMainActivityPath(args) {
+  const packageName = fs.readFileSync(
+    'app/src/main/AndroidManifest.xml',
+    'utf8'
+  ).match(/package="(.+?)"/)[1];
+  const mainActivity = args.mainActivity || 'MainActivity';
+  const variant = args.variant
+
+  if (variant) {
+    const variantKeys = variant.replace(/([A-Z])/g, ' $1').trim().toLowerCase().split(' ');
+    const productFlavor = variantKeys[0], buildType = variantKeys[1];
+    const packagePath = packageName.replace('.', '/');
+    const applicationId = fs.readFileSync(
+      `app/build/generated/source/buildConfig/${productFlavor}/${buildType}/${packagePath}/BuildConfig.java`,
```

Comment:
`productFlavor` is not a compulsory field so it may or may not exist. In cases where it doesn't the path to the `BuildConfig.java` file will be malformed. Eg:  (`app/build/generated/source/buildConfig//debug/com/example/BuildConfig.java`)